### ISSUE TITLE
ClassInfoList.getClassesImplementing() throws exception if underlying object is a class despite documentation stating the contrary

### ISSUE THREAD

**Issue by parttimenerd** [classgraph/classgraph#543]
The documentation and the code differ for [`ClassInfoList.getClassesImplementing()`](https://github.com/classgraph/classgraph/blob/046a86d5f74c140e9d010d39a44b02fe3b94c17c/src/main/java/io/github/classgraph/ClassInfo.java#L1687):
```java
    /**
     * Get the classes (and their subclasses) that implement this interface, if this is an interface.
     *
     * @return the list of the classes (and their subclasses) that implement this interface, if this is an
     *         interface, otherwise returns the empty list.
     */
    public ClassInfoList getClassesImplementing() {
        if (!isInterface()) {
            throw new IllegalArgumentException("Class is not an interface: " + getName());
        }
```
The code clearly throws an `IllegalArgumentException` for non-interfaces.

This propagates to [`ScanResult.getClassesImplementing()`](https://github.com/classgraph/classgraph/blob/046a86d5f74c140e9d010d39a44b02fe3b94c17c/src/main/java/io/github/classgraph/ScanResult.java#L1031) too which therefore has the same issue.

**Comment by lukehutch** [comment]
Thanks, good catch. Do you think this should throw an exception, or just return the empty list? The argument for throwing an exception is that it may help find bugs in a user's code, but maybe it is better never to throw exceptions, and fail gracefully instead.

**Comment by parttimenerd** [comment]
I would personally prefer the latter. My current use case (where this came up) is getting all implementing classes and subclasses for a ClassInfo object (not caring about the distinction between interfaces and classes). 

A preliminary search on GitHub for usages of the method shows that most usages are in the context where the `ClassInfo` object is expected to be an interface.

**Comment by lukehutch** [comment]
Fixed and released in 4.8.113. Thanks!
